[WIP] Fixedwing trim

bresch opened the pull request:
I started to add details about how to set the trim parameters for a fixedwing aircraft.

@hamishwillee Should I make a **Fixedwing** sub-category in **Advanced Configurations** and put the PID tuning and trimming guides (like it is now for VTOL)?

hamishwillee commented on en/advanced_config/pid_tuning_guide_fixedwing.md:
```diff
@@ -11,7 +11,7 @@
 >  - Roll and pitch tuning follow the same
 >    sequence. The only difference is that pitch is more sensitive to trim
->    offsets, so integrator gains need more attention to compensate this.
+>    offsets, so [trimming]() has to be done carefully and integrator gains need more attention to compensate this.
```
link missing

hamishwillee commented on en/advanced_config/trimming_guide_fixedwing.md:
```diff
@@ -0,0 +1,33 @@
+
+1. Trim the servos by physically adjusting the linkages lengths if possible and fine tune by trimming the PWM channels (use `PWM_MAIN/AUX_TRIMx`) on the bench to properly set the control surfaces to their theoretical position.
+
+2. Fly in stabilized mode at cruise speed and set the pitch setpoint offset (`FW_PSP_OFF`) to desired angle of attack. The required angle of attack at cruise speed corresponds to the pitch angle the airplane needs to fly at to keep constant altitude during wing-leveled flight. If you are using an airpeed sensor, also set the correct cruise airspeed ('FW_AIRSPD_TRIM`).
```
FW_AIRSPD_TRIM has mismatched quote marks (so does not render as code)

hamishwillee commented:
Hi @bresch 

> Should I make a Fixedwing sub-category in Advanced Configurations and put the PID tuning and trimming guides (like it is now for VTOL)?

Leave it for now - I'll look at grouping when this is merged (will probably do MC too).

> Should I link all the parameters to the parameter list page?

I usually link the first instance and make the other instances `Code`. You can leave this if you want and I will tidy in a subedit afterwards. 

> Add simple graph for the bilinear curve in advanced trimming

Yes.

I would start this at the very top with a single sentence or paragraph explaining what trimming is/is for.

bresch commented:
Hi @hamishwillee , thanks, I added the links to the parameter page and fixed the quote mark typo.

I'll add the graph and a description to explain what trimming is/is for when I have more time :)

hamishwillee commented:
Great. Just let me know when you want to merge.

tstastny commented on en/advanced_config/trimming_guide_fixedwing.md:
```diff
@@ -0,0 +1,33 @@
+
+- [FW_DTRIM_R/P/Y_VMIN/VMAX](../advanced_config/parameter_reference.md#FW_DTRIM_R_VMIN) is the roll/pitch/yaw trim value added to `TRIM_ROLL/PITCH/YAW` at min/max airspeed.
+
+- [FW_DTRIM_R/P_FLPS](../advanced_config/parameter_reference.md#FW_DTRIM_R_FLPS) is the roll/pitch trim value added to `TRIM_ROLL/PITCH/YAW` when the flaps are deployed.
```
@bresch only minor comment here, not sure if it should be explicitly mentioned that the DTRIM for flaps is applied in full only when flaps are deployed to full. So if a user has a flap scale set (say they are only using 50%) - they should not expect the entire DTRIM to be applied, but only 50% of the DTRIM -- this similarly applies to trimming this value in flight, full flaps should ideally be deployed when trying to find this DTRIM.

hamishwillee replied:
@bresch Did you address/consider/see this comment? If so, please mark resolved.

hamishwillee replied:
Ignore that, I see you did :-)

hamishwillee commented:
@bresch This is just a ping reminder :-)

bresch commented:
Graph and introduction are added.
I also corrected a typo in the DRIM parameter (#10668)

hamishwillee commented on en/advanced_config/trimming_guide_fixedwing.md:
```diff
@@ -38,4 +38,4 @@ Request access from dev team. -->
 A perfectly symmetrical airframe would only require pitch trim increments, but since a real airframe is never perfectly symmetrical, roll and yaw trims increments are also sometimes required. 
 
-> **Note** If a scale factor different from 1.0 is used for the flaps (parameter [FW_FLAPS_SCL](https://dev.px4.io/en/advanced/parameter_reference.html#FW_FLAPS_SCL), the trim increment added by the `FW_DTRIM_R/P_FLPS`is also scaled by the same factor.
+> **Note** If a scale factor different from 1.0 is used for the flaps (parameter [FW_FLAPS_SCL](https://dev.px4.io/en/advanced/parameter_reference.html#FW_FLAPS_SCL)), the trim increment added by the `FW_DTRIM_R/P_FLPS`is also scaled by the same factor.
```
There is also a local copy of the parameter reference in the user guide. Please change the link:
* FROM https://dev.px4.io/en/advanced/parameter_reference.html#FW_FLAPS_SCL 
* TO ../advanced_config/parameter_reference.md#FW_FLAPS_SCL

hamishwillee commented on en/advanced_config/trimming_guide_fixedwing.md:
```diff
@@ -24,10 +24,16 @@ The correct order to set those parameters is:
+Given that the downward pitch moment induced by an asymmetric airfoil increases witch airspeed and when the flaps are deployed, the aircraft needs to be re-trimmed according to the current measured airspeed and flaps position. For this purpose, a bilinear curve (see figure below) function of airspeed and a pitch trim increment function of the flaps state can be defined using the following paramet ...
 
-- [FW_DTRIM_R/P/Y_VMIN/VMAX](../advanced_config/parameter_reference.md#FW_DTRIM_R_VMIN) is the roll/pitch/yaw trim value added to `TRIM_ROLL/PITCH/YAW` at min/max airspeed.
+- [FW_DTRIM_R/P/Y_VMIN/VMAX](../advanced_config/parameter_reference.md#FW_DTRIM_R_VMIN) is the roll/pitch/yaw trim value added to `TRIM_ROLL/PITCH/YAW` at min/max airspeed (defined by [FW_AIRSPD_MIN](https://dev.px4.io/en/advanced/parameter_reference.html#FW_AIRSPD_MIN) and [FW_AIRSPD_MAX](https://dev.px4.io/en/advanced/parameter_reference.html#FW_AIRSPD_MAX)).
```
Link to local copy of parmeters

hamishwillee commented:
Looking pretty good. Let me know when you want final review and merge.

bresch commented:
I think we're good to go! :) Thanks for the reviews.

hamishwillee commented:
Thanks @bresch. Merged, moved the doc under "FW Config" and did minor tidy as discussed. You can see final version here: https://docs.px4.io/en/config_fw/trimming_guide_fixedwing.html
